src/foam/dao/CopyFromDAO.js

```diff
@@ -268,9 +268,7 @@ foam.CLASS({
               .setDelegate(sink != null ? sink : new ArraySink())
               .setOf(this.getOf())
               .build();
-            getDelegate().select_(x,
-              innerSink instanceof Count ? sink : decoratedSink,
-              skip, limit, adaptOrder(order), adaptPredicate(predicate));
+            getDelegate().select_(x, decoratedSink, skip, limit, adaptOrder(order), adaptPredicate(predicate));
```

Comment:
What's the reason for this change?

Reply:
Please add this comment here:

there is a case when we need the decorated sink even when the inner most sink is count
e.g, when we want to apply predicate after the adapt. AdapterSink -> PredicatedSink -> CountSink